File: A's/py/136A.py
def main() -> None:
    n: int = int(input())
    p = list(map(int, input().split()))

    # gave_to[i] will be the friend that i gave a gift to
    gave_to = [0] * (n + 1)
    for receiver, giver in enumerate(p, start=1):
        gave_to[giver] = receiver

    # output the mapping for friends 1 through n
    print(*gave_to[1:])

if __name__ == "__main__":
    main()

File: A's/cpp/136A.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int n; cin >> n;
    vector<int> p(n+1), gave_to(n+1);
    // p[i] = the friend who gave a gift to friend i
    for (int i = 1; i <= n; i++) {
        cin >> p[i];
    }
    // Invert: for each receiver i, giver = p[i], so giver gave to i
    for (int i = 1; i <= n; i++) {
        gave_to[p[i]] = i;
    }
    // Output: gave_to[i] = the friend to whom friend i gave a gift
    for (int i = 1; i <= n; i++) {
        cout << gave_to[i] << (i==n ? '\n' : ' ');
    }
    
    return 0;
}